src/main/java/net/pms/dlna/DLNAThumbnail.java

```diff
@@ -0,0 +1,263 @@
+import net.pms.util.CustomImageReader.ImageReaderResult;
+import net.pms.util.ImagesUtil;
+import net.pms.util.ImagesUtil.ImageFormat;
+import net.pms.util.ImagesUtil.ScaleType;
+
+/**
+ * This class is simply a byte array for holding an {@link ImageIO} supported
+ * image with some additional metadata.
+ *
+ * @see DLNAThumbnailInputStream
+ */
+public class DLNAThumbnail implements Serializable {
+
+	private static final long serialVersionUID = 8014825365045944137L;
+	private static final Logger LOGGER = LoggerFactory.getLogger(DLNAThumbnail.class);
+	private final byte[] bytes;
+	private final int width;
+	private final int height;
+	private final ImageFormat format;
+
+	/**
+	 * Creates a new {@link DLNAThumbnail} instance.
+	 *
+	 * @param bytes the source image in a supported format. Format support is
+	 * limited to that of {@link ImageIO}.
+	 *
+	 * @param width the width of the source image.
+	 * @param height the height of the source image.
+	 * @param format the {@link ImageFormat} of the source image.
+	 * @param copy whether the source image should be copied or shared.
+	 */
+	@SuppressFBWarnings("EI_EXPOSE_REP2")
+	public DLNAThumbnail(byte[] bytes, int width, int height, ImageFormat format, boolean copy) {
+		if (copy && bytes != null) {
+			this.bytes = new byte[bytes.length];
+			System.arraycopy(bytes, 0, this.bytes, 0, bytes.length);
+		} else {
+			this.bytes = bytes;
+		}
+		this.width = width;
+		this.height = height;
+		this.format = format;
+	}
+
+	/**
+	 * Converts an image to a {@link DLNAThumbnail}. Format support is limited
+	 * to that of {@link ImageIO}.
+	 *
+	 * @param imageByteArray the source image in a supported format.
+	 * @return The populated {@link DLNAThumbnail} or {@code null} if the
+	 *         source image could not be parsed.
+	 */
+	public static DLNAThumbnail toThumbnail(byte[] imageByteArray) {
+		return toThumbnail(imageByteArray, 0, 0, null, ImageFormat.SOURCE);
+	}
+
+	/**
+	 * Converts an image to a {@link DLNAThumbnail}. Format support is limited
+	 * to that of {@link ImageIO}.
+	 *
+	 * @param imageByteArray the source image in a supported format.
+	 * @param width the new width or 0 to disable scaling.
+	 * @param height the new height or 0 to disable scaling.
+	 * @param scaleType the {@link ScaleType} to use when scaling.
+	 * @param outputFormat the {@link ImageFormat} to generate or
+	 *                     {@link ImageFormat#SOURCE} to preserve source format.
+	 * @return The populated {@link DLNAThumbnail} or {@code null} if the
+	 *         source image could not be parsed.
+	 */
+	public static DLNAThumbnail toThumbnail(byte[] imageByteArray, int width, int height, ScaleType scaleType, ImageFormat outputFormat) {
+		return imageByteArray != null ? toThumbnail(new ByteArrayInputStream(imageByteArray), width, height, scaleType, outputFormat) : null;
+	}
+
+	/**
+	 * Converts an image to a {@link DLNAThumbnail}. Format support is limited
+	 * to that of {@link ImageIO}.
+	 * <p><b>
```

Comment:
@Sami32 Anything inside ```/**``` and ```*/``` is per definition [JavaDoc](https://en.wikipedia.org/wiki/Javadoc). It is rendered by IDE's but can also be used to generate [standalone JavaDocs](https://commons.apache.org/proper/commons-lang/javadocs/api-release/index.html).